### ISSUE TITLE
Mismatched static class method overloads (> 100 char MnglName?)

### ISSUE THREAD

**Issue by swebb2066** [lvc/abi-compliance-checker#133]
Incorrectly reports register mismatch (as High Severity) in log4cxx library when using the wrong overload:
```
bool StringHelper::equalsIgnoreCase(const LogString& s1, const logchar* upper, const logchar* lower);
bool StringHelper::equalsIgnoreCase(const LogString& s1, const LogString& upper, const LogString& lower)

```
extract from dump showing mangled 

[example.tar.gz](https://github.com/user-attachments/files/18515370/example.tar.gz)

names:
```
                            '562000' => {
                                          'Class' => '561899',
                                          'Header' => 'stringhelper.h',
                                          'Line' => '39',
                                          'MnglName' => '_ZN7log4cxx7helpers12StringHelper16equalsIgnoreCaseERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPKcSB_',
                                          'Param' => {
                                                       '0' => {
                                                                'name' => 's1',
                                                                'type' => '210597'
                                                              },
                                                       '1' => {
                                                                'name' => 'upper',
                                                                'type' => '575792'
                                                              },
                                                       '2' => {
                                                                'name' => 'lower',
                                                                'type' => '575792'
                                                              }
                                                     },
                                          'Reg' => {
                                                     '0' => 'rdi',
                                                     '1' => 'rsi',
                                                     '2' => 'rdx'
                                                   },
                                          'Return' => '174077',
                                          'ShortName' => 'equalsIgnoreCase',
                                          'Source' => 'stringhelper.cpp',
                                          'SourceLine' => '31',
                                          'Static' => 1
                                        },
                            '562036' => {
                                          'Class' => '561899',
                                          'Header' => 'stringhelper.h',
                                          'Line' => '41',
                                          'MnglName' => '_ZN7log4cxx7helpers12StringHelper16equalsIgnoreCaseERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES9_S9_',
                                          'Param' => {
                                                       '0' => {
                                                                'name' => 's1',
                                                                'type' => '210597'
                                                              },
                                                       '1' => {
                                                                'name' => 'upper',
                                                                'type' => '210597'
                                                              },
                                                       '2' => {
                                                                'name' => 'lower',
                                                                'type' => '210597'
                                                              }
                                                     },
                                          'Reg' => {
                                                     '0' => 'rbx',
                                                     '1' => 'r10',
                                                     '2' => 'rbp'
                                                   },
                                          'Return' => '174077',
                                          'ShortName' => 'equalsIgnoreCase',
                                          'Source' => 'stringhelper.cpp',
                                          'SourceLine' => '46',
                                          'Static' => 1
                                        },
 
```

**Comment by swebb2066** [comment]
The issues arose when comparing a dump from gcc 9 with a dump from gcc 11
